core/src/com/unciv/models/translations/TranslationFileWriter.kt

```diff
@@ -17,99 +18,130 @@ import java.lang.reflect.Field
 
 object TranslationFileWriter {
 
+    private const val specialNewLineCode = "# This is an empty line "
     const val templateFileLocation = "jsons/translations/template.properties"
-    private val generatedStrings = mutableMapOf<String,MutableSet<String>>()
+    private const val languageFileLocation = "jsons/translations/%s.properties"
 
-    private fun writeByTemplate(language:String, translations: HashMap<String, String>){
+    fun writeNewTranslationFiles(translations: Translations) {
 
-        val templateFile = Gdx.files.internal(templateFileLocation)
-        val linesFromTemplates = mutableListOf<String>()
-        linesFromTemplates.addAll(templateFile.reader().readLines())
+        val percentages = generateTranslationFiles(translations)
+        writeLanguagePercentages(percentages)
+
+        // try to do the same for the mods
+        for(modFolder in Gdx.files.local("mods").list())
+            generateTranslationFiles(translations, modFolder)
+            // write percentages is not needed: for an individual mod it makes no sense
+
+    }
 
-        generateStringsFromJSONs()
+    private fun generateTranslationFiles(translations: Translations, modFolder: FileHandle? = null): HashMap<String, Int> {
+        // read the template
+        val templateFile = if (modFolder != null) modFolder.child(templateFileLocation)
+                            else Gdx.files.internal(templateFileLocation)
+        val linesFromTemplates = mutableListOf<String>()
+        if (templateFile.exists())
+            linesFromTemplates.addAll(templateFile.reader().readLines())
+        // read the JSON files
+        val generatedStrings = generateStringsFromJSONs(modFolder)
         for (key in generatedStrings.keys) {
             linesFromTemplates.add("\n#################### Lines from $key.json ####################\n")
             linesFromTemplates.addAll(generatedStrings.getValue(key))
         }
 
-        val stringBuilder = StringBuilder()
-        for(line in linesFromTemplates){
-            if(!line.contains(" = ")){ // copy as-is
-                stringBuilder.appendln(line)
-                continue
+        var countOfTranslatableLines = 0
+        val countOfTranslatedLines = HashMap<String, Int>()
+        // iterate through all available languages
+        for (language in translations.getLanguages()) {
+            var translationsOfThisLanguage = 0
+            val stringBuilder = StringBuilder()
+
+            for (line in linesFromTemplates) {
+                if (!line.contains(" = ")) {
+                    // small hack to insert empty lines
+                    if (line.startsWith(specialNewLineCode))
```

Comment:
Why not just have it an empty line, as it has been so far?
It'll enter if(!line.contains(" = "), and be copied as-is - that is to say, empty

Reply:
Ahhh nevermind I see what you did below